open-banking-accelerator/components/com.wso2.openbanking.accelerator.common/src/main/java/com/wso2/openbanking/accelerator/common/util/ErrorConstants.java

```diff
@@ -221,9 +245,56 @@ public class ErrorConstants {
             " retrieval request";
     public static final String INVALID_CONSENT_ID = "Invalid Consent Id found in the request";
     public static final String CONSENT_ID_NOT_FOUND = "Consent ID not available in consent data";
-
     public static final String FIELD_INVALID_DATE = "OB.Field.InvalidDate";
     public static final String EXPIRED_DATE_ERROR = "The ExpirationDateTime value has to be a future date.";
+    // vrp
+    public static final String MAXIMUM_INDIVIDUAL_AMOUNT_IS_MISSING = "Mandatory parameter MaximumIndividualAmount" +
+            " Amount is missing in the payload.";
+    public static final String MAXIMUM_INDIVIDUAL_AMOUNT_CURRENCY_IS_MISSING = "Mandatory parameter" +
+            "Currency in MaximumIndividualAmount is missing in the payload";
+    public static final String INVALID_AMOUNT = "Missing required parameter Amount";
+    public static final String INVALID_CURRENCY = "Missing required parameter Currency";
+    public static final String INVALID_PERIOD_ALIGNMENT = "Invalid value for period alignment in PeriodicLimits";
+    public static final String MISSING_PERIOD_TYPE = "Missing required parameter Period type";
+    public static final String INVALID_VALID_TO_DATE = "Valid to Date specified in the request is invalid";
+
+    // new error constants
+
+    public static final String INVALID_INITIATION_PAYLOAD = "Consent validation failed due to invalid" +
+            " initiation payload";
+    public static final String INVALID_PARAMETER = "Parameter passed in is null , " +
+            "empty or not a JSONObject";
+    public static final String INVALID_CLIENT_ID_MATCH = "Consent validation failed due to client ID mismatch";
+    public static final String INVALID_PARAMETER_DEBTOR_ACC = "Parameter passed in is null , " +
+            "empty or not a JSONObject in debtor account";
+    public static final String INVALID_PARAMETER_CREDITOR_ACC = "Parameter passed in is null , " +
+            "empty or not a JSONObject in debtor account";
+    public static final String INVALID_PARAMETER_AMOUNT = "Parameter passed in is null , " +
+            "empty or not a JSONObject in amount";
+    public static final String INVALID_PARAMETER_INITIATION = "Parameter passed in is null , " +
+            "empty or not a JSONObject in initiation";
+    public static final String INVALID_PARAMETER_CONTROL_PARAMETERS = "Parameter passed in is null , " +
+            "empty or not a JSONObject in control parameter";
+    public static final String INVALID_PARAMETER_MAXIMUM_INDIVIDUAL_AMOUNT = "Parameter passed in is null , " +
+            "empty or not a JSONObject in maximum individual amount";
+    public static final String INVALID_DATE_TIME_FORMAT = "Date and Time is not in correct JSON " +
+            "ISO-8601 date-time format";
+    public static final String INVALID_PARAMETER_PERIODIC_LIMITS = "Parameter passed in is null , " +
+            "empty or not a JSONArray";
```

Comment:
Shall we use a single error message and pass the parameter name to build the message without defining similar set of strings? You can use the validation function itself to build the proper error message

Reply:
Fixed in [a575078]